src/main/java/org/folio/circulation/domain/FeeFineOwnerRepository.java

```diff
@@ -21,19 +21,20 @@ public FeeFineOwnerRepository(Clients clients) {
   }
 
   public CompletableFuture<Result<FeeFineOwner>> findOwnerForServicePoint(String servicePointId) {
-    return CqlQuery.notEqual(SERVICE_POINT_OWNER_KEY, EMPTY_ARRAY)
-      .after(query -> feeFineOwnerStorageClient.getMany(query, OWNERS_PAGE_LIMIT))
+    return CqlQuery.match(SERVICE_POINT_OWNER_KEY, servicePointId)
+      .after(query -> feeFineOwnerStorageClient.getMany(query, one()))
       .thenApply(r -> r.next(this::mapResponseToOwners))
       .thenApply(r -> r.map(MultipleRecords::getRecords))
-      .thenApply(r -> r.map(owners -> owners.stream()
-        .filter(owner -> owner.getServicePoints().stream().anyMatch(servicePointId::equals))
-        .findAny()
-        .orElse(null))
-      );
+      .thenApply(r -> r.map(this::firstOrNull));
+  }
+
+  private <T> T firstOrNull(Collection<T> collection) {
+    return collection.stream()
+      .findFirst()
+      .orElse(null);
```

Comment:
Please could this be moved to a helper class for collections, so that it can be more easily discovered and reused.

Reply:
Moved to `org.folio.circulation.support.utils.CollectionUtil` class.